src/socket/mod.rs

```diff
@@ -218,14 +232,14 @@ impl Socket {
 
         while let Some(Ok(response)) = iter.next() {
             match response.nl_type {
-                    Nlmsg::Error => panic!("Error"),
-                    Nlmsg::Done => break,
-                    _ => {
-                        let  handle = response.nl_payload.get_attr_handle();
-                        return Ok(Bss::default().parse(handle))
-                    }
+                Nlmsg::Error => panic!("Error"),
```

Comment:
I know you just reformatted this code but if it isn't a huge lift to refactor out some of these panics we should. As is I'm a little worried about using this crate in our code.